plugins/python/src/main/kotlin/io/paddle/plugin/python/extensions/Requirements.kt

```diff
@@ -15,14 +16,17 @@ val PaddleProject.requirements: Requirements
     get() = checkNotNull(extensions.get(Requirements.Extension.key)) { "Could not load extension Requirements for project $routeAsString" }
 
 class Requirements(val project: PaddleProject, val descriptors: MutableList<Descriptor>) : Hashable {
-
     val resolved: Collection<PyPackage> by lazy {
         val installedPackages = project.environment.venv.pyPackages
         val resolvedPackages = try {
             PipResolver.resolve(project)
         } catch (e: PipResolver.RetrySignal) {
-            project.terminal.warn("Retrying resolve...")
-            PipResolver.resolve(project)
+            if (project.pythonRegistry.autoRetry) {
+                project.terminal.warn("Retrying resolve...")
+                PipResolver.resolve(project)
+            } else {
+                throw Task.ActException("Retry is needed, but autoRetry option is set to false")
```

Comment:
"Resolution failed due to issues with local pip's cache. You can either activate `autoRetry` option in the settings/registry, or disable pip's default caching for Paddle resolve/install commands (check `noCacheDir option`)."

Reply:
Let's explain the errors in a way which user will understand :)